### PR TITLE
[C#] Extract concepts from v2 exercise: high-scores

### PR DESCRIPTION
**Pull request by fpetru**
Ref #721 - work in progress

### CODE REVIEW

**Comment by fpetru** [comment]
please help to review this PR. thank you

File: languages/csharp/reference/exercise-concepts/high-scores.md [hunk 1]
```diff
@@ -0,0 +1,29 @@
...
+    - use `{` and `}` to denote scoping
+    - visibility: making tested method and tested class `public`
+    - imports: import types through `using` statements (collections, LINQ)
+
+### Types
+- numbers: 
+    - signed integers: used for score values
+    - lists: strongly typed list of `int` objects 
+
+### Functional
```
Comment (robkeim) [inline languages/csharp/reference/exercise-concepts/high-scores.md]: I might be nit picking here, but this exercise is solvable without using LINQ (although the LINQ solution is shorter). You can simplify iterate to find the max, sort to find the top three, etc.

Given that, I wonder if it makes sense to move this to describe one approach and loops as another one.

Comment (fpetru) [reply languages/csharp/reference/exercise-concepts/high-scores.md]: Sure, it is solvable in both ways - I will include also the non-LINQ approach in the concepts list.

File: languages/csharp/reference/exercise-concepts/high-scores.md [hunk 2]
```diff
@@ -0,0 +1,33 @@
...
+    - the tested methods are defined in a class
+- methods: 
+    - methods arguments: the time is passed as an argument
+    - return values: returning a value from a method
+
+### General
+- scoping
+    - use `{` and `}` to denote scoping
+    - visibility: making tested method and tested class `public`
+    - visibility: making the list that contains the scores `private` and potentially `readonly`
```
Comment (ErikSchierboom) [inline languages/csharp/reference/exercise-concepts/high-scores.md]: Isn't this a duplicate Concept? I'd prefer to only list each Concept once, so maybe you can combine the two lines?

File: languages/csharp/reference/exercise-concepts/high-scores.md [hunk 3]
```diff
@@ -0,0 +1,33 @@
+# Concepts of high-scores
+
+[Example implementation](https://github.com/exercism/csharp/blob/master/exercises/high-scores/Example.cs)
+
+## General concepts
```
Comment (ErikSchierboom) [inline languages/csharp/reference/exercise-concepts/high-scores.md]: Some general concepts I think are missing:

- assignment
- immutability: the readonly modifier
- enumerables: lists can can be iterated over as an enumerable (using Linq)

https://github.com/exercism/v3/blob/master/languages/csharp/reference/exercise-concepts/book-store.md#approach-switch

Comment (archrisV) [reply languages/csharp/reference/exercise-concepts/high-scores.md]: I tried to address your issues with those three categories. Hopefully is this version better.

Comment (ErikSchierboom) [reply languages/csharp/reference/exercise-concepts/high-scores.md]: Yep it's nice. Just a FYI: usually it is up to the PR submitter to fix these things, which can help the PR submitted better keep track of what is happening.

Comment (archrisV) [reply languages/csharp/reference/exercise-concepts/high-scores.md]: I'm sorry, I didn't know that. I tried to do it as a pull request, but it gave an error for an unknown URL. I'll keep this in mind for future reference.

Comment (ErikSchierboom) [reply languages/csharp/reference/exercise-concepts/high-scores.md]: Sure. No worries.

File: languages/csharp/reference/exercise-concepts/high-scores.md (hunk 1 above)
Comment (fpetru) [reply languages/csharp/reference/exercise-concepts/high-scores.md]: Thanks, fixed

File: languages/csharp/reference/exercise-concepts/high-scores.md (hunk 2 above)
Comment (fpetru) [reply languages/csharp/reference/exercise-concepts/high-scores.md]: Yes, cleaned

File: languages/csharp/reference/exercise-concepts/high-scores.md (hunk 3 above)
Comment (fpetru) [reply languages/csharp/reference/exercise-concepts/high-scores.md]: Thanks @ErikSchierboom & @archrisV - added these, and a couple of additional concepts

**Comment by fpetru** [comment]
@ErikSchierboom - please help me on the merged version of the pull request

**Comment by ErikSchierboom** [comment]
@fpetru Could you rebase this PR on the latest master?

**Comment by ErikSchierboom** [comment]
@iHiD Could you merge?

**Pull request merged**